packages/frontend/src/components/accounts/batch_import_accounts/index.js

```diff
@@ -38,28 +39,132 @@ const CustomContainer = styled.div`
+
+            > span {
+                margin-left: 10px;
+                word-break: break-word;
+                color: #006ADC;
+            }
+
+            b {
+                color: #272729;
+            }
+        }
+
+        > button {
+            width: 100%;
+        }
+    }
+`;
+
 export const IMPORT_STATUS = {
   PENDING: 'pending',
   SUCCESS: 'success',
   UP_NEXT: 'waiting',
   FAILED: 'error'
 };
 
-const BatchImportAccounts = ({ accountIdToKeyMap, onCancel }) => {
+const EnterPublicKeyForm = ({ onCancel, onPublicKey }) => {
+    const [value, setValue] = useState('');
+    const handleInputChange = useCallback(({ currentTarget }) => {
+        setValue(currentTarget.value);
+    }, []);
+
+    const handleSubmit = useCallback(() => {
+        onPublicKey(value);
+    }, [value]);
+
+    return (
+        <Modal
+            isOpen
+            disableClose
+            modalSize='md'
+            style={{ maxWidth: '496px' }}
+        >
+            <PublicKeyFormContainer>
+                <h3><Translate id='batchImportAccounts.enterKeyForm.title' /></h3>
+                <p><Translate id='batchImportAccounts.enterKeyForm.desc' /></p>
+                <Translate>
+                    {({ translate }) => (
+                        <>
+                            <input
+                                placeholder={translate('batchImportAccounts.enterKeyForm.placeholder')}
+                                onChange={handleInputChange}
+                                value={value}
+                                autoCapitalize='off'
+                                spellCheck='false'
+                                autoFocus
+                            />
+                        </>
+                    )}
+                </Translate>
+                <FormButton
+                    disabled={value.length === 0}
+                    type='submit'
+                    onClick={handleSubmit}>
+                    <Translate id='batchImportAccounts.enterKeyForm.importCaption' />
+                </FormButton>
+                <FormButton
+                    className='link'
+                    onClick={onCancel}>
+                    <Translate id='button.cancel' />
+                </FormButton>
+            </PublicKeyFormContainer>
+        </Modal>
+    );
+};
+
+const ImportAccounts = ({
+    accountId,
+    secret,
+    onCancel,
+    ledgerHd,
+}) => {
     const availableAccountsIsLoading = useSelector(selectAvailableAccountsIsLoading);
     const availableAccounts = useSelector(selectAvailableAccounts);
     const accountUrlReferrer = useSelector(selectAccountUrlReferrer);
 
     const [state, dispatch] = useImmerReducer(reducer, {
-        accounts: Object.entries(accountIdToKeyMap).map(
-            ([accountId, { key, ledgerHdPath }]) => ({
-                accountId,
-                status: null,
-                key,
-                ledgerHdPath,
-            })
-        ),
+        accounts: [{
```

Comment:
It working for single account.
We not supporting multi-transfer now.

But code doesn't change, in the future we will support.

Reply:
This breaks the work we are doing to allow for mulltiple accounts export transfer.  This is a degradation of experience for users and also the code doesn't make sense here.

Reply:
@agileurbanite As this feature has never been released to users, there is no possibility of degradating experience for them (as there is none yet).

Implementing batch-import as was visioned before is impossible without imposing severe security risks to users, which is a no-go.

Aim of this changes is to allow users to import accounts by one by one, which is a good start to give majority of users (who is using single or small number of accounts) a way to transfer account.

Next step would be to enable multi-importing as was proposed initially: we know how to do it, but it will require a week of time or so. Which is out of our deadlines.

Anyway, I see a possible improvement here is to create another route called "single-account-transfer" - which will re-create flow without multiple options, to simplify flow and simplify future integration for real "batch-import"